server/src/bootstrap/passport/local.strategy.ts

```diff
@@ -20,9 +22,18 @@ export const localStrategy = (
           }
           const verifyHashResult = await verifyHash(otp, token.hashedOtp)
           if (!verifyHashResult) {
-            return done(null, false, {
-              message: 'OTP is invalid. Please try again.',
-            })
+            if (token.attempts + 1 >= MAX_OTP_ATTEMPTS) {
+              await Token.destroy({ where: { contact: email } })
+              return done(null, false, {
+                message:
+                  'You have hit the max number of attempts. Please request for a new OTP.',
+              })
+            } else {
+              await token.increment('attempts')
+              return done(null, false, {
+                message: 'OTP is invalid. Please try again.',
+              })
+            }
```

Comment:
This doesn't really stop a determined attacker though - asking for a new OTP is cheap. you have to figure a better way to deter the attacker, maybe with rate-limiting